core/src/main/scala/kafka/log/ProducerIdMapping.scala

```diff
@@ -24,13 +24,28 @@ import java.nio.file.Files
 import kafka.common.{KafkaException, TopicAndPartition}
 import kafka.utils.{Logging, nonthreadsafe}
 import org.apache.kafka.common.TopicPartition
-import org.apache.kafka.common.errors.{DuplicateSequenceNumberException, InvalidSequenceNumberException, ProducerFencedException}
+import org.apache.kafka.common.errors.{DuplicateSequenceNumberException, OutOfOrderSequenceException, ProducerFencedException}
 import org.apache.kafka.common.protocol.types._
+import org.apache.kafka.common.record.LogEntry
 import org.apache.kafka.common.utils.Utils
 
 import scala.collection.mutable
 
-private[log] case class PidEntry(seq: Int, epoch: Short, offset: Long)
+case class PidEntry(lastSeq: Int,
```

Comment:
This should still be `private[log]`. Probably the reason it doesn't work is that `ProducerIdMappingTest` is in the wrong package. Instead of `unit.kafka.idempotence`, it should be `kafka.log`.